volume/drivers/pwx/connection.go

```diff
@@ -0,0 +1,210 @@
+package pwx
+
+import (
+	"fmt"
+	"github.com/libopenstorage/openstorage/pkg/grpcserver"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type ConnectionParamsBuilder struct {
+	Config  *ConnectionParamsBuilderConfig
+	kubeOps core.Ops
+}
+
+type ConnectionParamsBuilderConfig struct {
+	// DefaultServiceName is the default name of the Portworx service
+	DefaultServiceName string
+	// DefaultServiceNamespaceName is the kubernetes namespace in which Portworx daemon set runs
+	DefaultServiceNamespaceName string
+	// DefaultRestPortName is name of Porx legacy REST API port in service
+	DefaultRestPortName string
+	// DefaultSDKPortName is name of Porx SDK/iSDK API port in service
+	DefaultSDKPortName string
+
+	// Environment variables names to get  values config properties
+	EnableTLSEnv       string
+	NamespaceNameEnv   string
+	ServiceNameEnv     string
+	CaCertSecretEnv    string
+	CaCertSecretKeyEnv string
+}
+
+var DefaultParams = &ConnectionParamsBuilderConfig{
+	DefaultServiceName:          "portworx-service",
+	DefaultServiceNamespaceName: "kube-system",
+	DefaultRestPortName:         "px-api",
+	DefaultSDKPortName:          "px-sdk",
+	EnableTLSEnv:                "PX_ENABLE_TLS",
+	NamespaceNameEnv:            "PX_NAMESPACE",
+	ServiceNameEnv:              "PX_SERVICE_NAME",
+	CaCertSecretEnv:             "PX_CA_CERT_SECRET",
+	CaCertSecretKeyEnv:          "PX_CA_CERT_SECRET_KEY",
+}
+
+func NewConnectionParamsBuilder(ops core.Ops, params *ConnectionParamsBuilderConfig) *ConnectionParamsBuilder {
+	if params == nil {
```

Comment:
nit: This works, but when reading the code that uses this, the reader does not know what `nil` will be interpreted as. Instead, there could be a `NewConnectionParamsBuilderConfig()` and a `NewConnectionParamsBuilderDefaultConfig()`. When the caller calls `NewConnectionParamsBuilderDefaultConfig()` they can use that to pass into this function. 

That way you do not have to silently set the values if it is not passed, and instead return an error.

Reply:
That is a good idea.